### PR TITLE
Implement RC-specific quorum when round higher than a threshold

### PR DESCRIPTION
**Pull request by R-Santev**
# Description
Faster halt resolution when validators
Require a smaller quorum for round change if the round is above 5 (goes there for ~5 minutes). Let's say 30% of the voting power thinks the right round is 12, other validators are fine to go on this round. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### CODE REVIEW

**Review by Vitomir2 — APPROVED** [review]
LGTM

File: core/drop_test.go [hunk 1]
```diff
@@ -80,6 +80,93 @@ func TestDropAllAndRecover(t *testing.T) {
...
+					},
+					baseRoundTimeout: DefaultBaseRoundTimeout,
+					validatorManager: NewValidatorManager(backend, mockLogger{}),
+				}
+			}
+		},
+	)
+
+	// Progress the chain to claim it works ok
+	err := cluster.progressToHeight(180*time.Second, 1)
```
Comment (Vitomir2) [inline core/drop_test.go:164]:
```suggestion
	err := cluster.progressToHeight(180 * time.Second, 1)
```

File: core/validator_manager.go [hunk 2]
```diff
@@ -26,6 +31,10 @@ type ValidatorManager struct {
 	// quorumSize represents quorum for the height specified in the current View
 	quorumSize *big.Int
 
+	// rcMinQuorum represents the voting power needed to create a round change certificate
+	// when round is above rcMinQuorumThreshold for the height specified in the current View
```
Comment (Vitomir2) [inline core/validator_manager.go:35]:
```suggestion
	// when the round is above rcMinQuorumThreshold for the height specified in the current View
```

**Review by Vitomir2 — APPROVED** [review]

File: core/drop_test.go (hunk 1 above)
Comment (R-Santev) [reply core/drop_test.go:164]: the linter formats it that way

**Review by Vitomir2 — APPROVED** [review]
LGTM

**Pull request merged**